fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java

```diff
@@ -327,9 +331,33 @@ public ShowResultSet execute() throws AnalysisException, DdlException {
 
     private void handleShowAuthentication() {
         final ShowAuthenticationStmt showAuthenticationStmt = (ShowAuthenticationStmt) stmt;
-        List<List<String>> rows = GlobalStateMgr.getCurrentState().getAuth().getAuthenticationInfo(
-                showAuthenticationStmt.getUserIdent());
-        resultSet = new ShowResultSet(showAuthenticationStmt.getMetaData(), rows);
+        if (connectContext.getGlobalStateMgr().isUsingNewPrivilege()) {
+            AuthenticationManager authenticationManager = GlobalStateMgr.getCurrentState().getAuthenticationManager();
+            List<List<String>> userAuthInfos = Lists.newArrayList();
+
+            Map<UserIdentity, UserAuthenticationInfo> authenticationInfoMap = new HashMap<>();
+            if (showAuthenticationStmt.getUserIdent() == null) {
```

Comment:
When null, we should check showAuthenticationStmt.isAll(), if is true, return all users info, if not, only return the auth info of current user, i think the original logic in `Auth` is wrong.

BTW, the privilege check of visitShowAuthenticationStatement() in PrivilegeCheckerV2 is incorrect, plz also modify that function.

![image](https://user-images.githubusercontent.com/107105845/210958226-4a2e35f1-dfc8-4d52-bba1-4d53beabae11.png)